typescript/web/src/hooks/use-image-navigation.fixtures.ts

```diff
@@ -0,0 +1,81 @@
+import { GET_ALL_IMAGES_OF_A_DATASET_QUERY } from ".";
+import {
+  GetAllImagesOfADatasetQuery,
+  GetAllImagesOfADatasetQueryVariables,
+  GetAllImagesOfADatasetQuery_dataset_images,
+} from "../graphql-types/GetAllImagesOfADatasetQuery";
+import {
+  ApolloMockResponse,
+  ApolloMockResponses,
+} from "../utils/tests/apollo-mock";
+import { BASIC_DATASET_DATA } from "../utils/tests/data.fixtures";
+import { CURRENT_IMAGE } from "./test-utils";
+
+export const IMAGE1 = {
+  id: "929f5445-812c-4679-86bd-0f0748679f38",
+  url: "",
+  thumbnail200Url: "",
+};
+
+export const IMAGE2 = {
+  id: "51e00f79-d178-4ea1-8bb5-b0fa647f9d19",
+  url: "",
+  thumbnail200Url: "",
+};
+
+export const IMAGE3 = {
+  id: "8067a965-d017-4c4b-a82d-98cc803d1020",
+  url: "",
+  thumbnail200Url: "",
+};
+
+const CURRENT_IMAGE_IS_FIRST = [CURRENT_IMAGE, IMAGE1, IMAGE2];
+const CURRENT_IMAGE_IS_SECOND = [IMAGE1, CURRENT_IMAGE, IMAGE2];
+const CURRENT_IMAGE_IS_LAST = [IMAGE1, IMAGE2, CURRENT_IMAGE];
+const CURRENT_IMAGE_NOT_PRESENT = [IMAGE1, IMAGE2, IMAGE3];
+
+const getImagesMockGenerator = (
+  images: GetAllImagesOfADatasetQuery_dataset_images[]
+): ApolloMockResponse<
+  GetAllImagesOfADatasetQuery,
+  GetAllImagesOfADatasetQueryVariables
+> => {
+  return {
+    request: {
+      query: GET_ALL_IMAGES_OF_A_DATASET_QUERY,
+      variables: {
+        slug: BASIC_DATASET_DATA.slug,
+        workspaceSlug: BASIC_DATASET_DATA.workspace.slug,
+      },
+    },
+    result: {
+      data: {
+        dataset: {
+          id: BASIC_DATASET_DATA.id,
+          images,
+        },
+      },
+    },
+    nMatches: Number.POSITIVE_INFINITY,
+  };
+};
+
+export const THREE_IMAGES_MOCKS: ApolloMockResponses = [
+  getImagesMockGenerator(CURRENT_IMAGE_IS_SECOND),
+];
+
+export const IMAGE_IS_FIRST_MOCKS: ApolloMockResponses = [
+  getImagesMockGenerator(CURRENT_IMAGE_IS_FIRST),
+];
+
+export const IMAGE_IS_LAST_MOCKS: ApolloMockResponses = [
+  getImagesMockGenerator(CURRENT_IMAGE_IS_LAST),
+];
+
+export const NO_IMAGES_MOCKS: ApolloMockResponses = [
+  getImagesMockGenerator([]),
+];
+
+export const CURRENT_NOT_IN_IMAGES_MOCKS: ApolloMockResponses = [
+  getImagesMockGenerator(CURRENT_IMAGE_NOT_PRESENT),
+];
```

Comment:
Can you add an intermediary `createImagesMocks` function which would return directly the array?

Reply:
Which array ?

Reply:
The `[` and `]` surrounding every value declared in the block I've mentioned.